serving/src/main/java/feast/serving/configuration/JobServiceConfig.java

```diff
@@ -16,48 +16,64 @@
  */
 package feast.serving.configuration;
 
+import feast.core.StoreProto;
 import feast.core.StoreProto.Store.StoreType;
 import feast.serving.FeastProperties;
 import feast.serving.service.JobService;
 import feast.serving.service.NoopJobService;
 import feast.serving.service.RedisBackedJobService;
 import feast.serving.specs.CachedSpecService;
-import java.util.Map;
+
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.RedisURI;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.codec.ByteArrayCodec;
+import io.lettuce.core.resource.ClientResources;
+import org.springframework.beans.factory.ObjectProvider;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import redis.clients.jedis.JedisPool;
-import redis.clients.jedis.JedisPoolConfig;
+
+import java.util.Map;
 
 @Configuration
 public class JobServiceConfig {
 
-  public static final String DEFAULT_REDIS_MAX_CONN = "8";
-  public static final String DEFAULT_REDIS_MAX_IDLE = "8";
-  public static final String DEFAULT_REDIS_MAX_WAIT_MILLIS = "50";
+  @Bean
+  Map<String, String> jobRedisConf(FeastProperties feastProperties) {
```

Comment:
Are all of these Redis Beans executed on every startup, even if Redis isn't the backing store? Are we annotating these methods as `Beans` because we want dependency injection? Could we perhaps only make the `jobService` the only Bean and move the store configuration out of the `JobServiceConfig` class?

Reply:
You are right. the configuration map doesn't explicitly say it's a redis conf. We can just do inside RedisBacked.